### ISSUE TITLE
CentOS 8-Stream install: invalid conmon path: stat /usr/libexec/crio/conmon: no such file or directory

### ISSUE THREAD

**Issue by R37ribution** [cri-o/cri-o#4653]
Looks like a repeat of #3818 as the workaround described in https://github.com/cri-o/cri-o/issues/3818#issuecomment-650899132 `sudo ln -s $(which conmon) /usr/libexec/crio/conmon` resolved the issue and the cri-o.service started.

Please note that this was an RPM taken from RedHat's site and I'm sorry but I'm not sure if this is the right place for this issue. The version matched that which is in this repo and I could not find an appropriate place on bugzilla for CRI-O.

**Steps to reproduce the issue:**
1. `yum install ./cri-o-1.20.1-2.rhaos4.7.git0c3216a.el8.x86_64.rpm`
2. `sudo systemctl enable cri-o --now`
3. `cri-o.service` does not start (see journald logs below)

**Describe the results you received:**
after installing cri-o package, cri-o.service does not start

**Describe the results you expected:**
cri-o.service should start

**Additional information you deem important (e.g. issue happens only occasionally):**
Reproducible every time

**Output of `crio --version`:**

```
hostname ~$ crio --version
INFO[0000] Starting CRI-O, version: 1.20.1-2.rhaos4.7.git0c3216a.el8, git: ()
crio version 1.20.1-2.rhaos4.7.git0c3216a.el8
Version:    1.20.1-2.rhaos4.7.git0c3216a.el8
GoVersion:  go1.15.5
Compiler:   gc
Platform:   linux/amd64
Linkmode:   dynamic
hostname ~$

hostname ~$ cat /etc/os-release
NAME="CentOS Stream"
VERSION="8"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="8"
PLATFORM_ID="platform:el8"
PRETTY_NAME="CentOS Stream 8"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:8"
HOME_URL="https://centos.org/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux 8"
REDHAT_SUPPORT_PRODUCT_VERSION="CentOS Stream"
hostname ~$ uname -r
4.18.0-277.el8.x86_64
hostname ~$
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
CentOS 8-Stream

journald logs from the `cri-o.service`:
```
Mar 11 20:46:18 hostname systemd[1]: Starting CRI-O Auto Update Script...
Mar 11 20:46:18 hostname crio[13892]: time="2021-03-11T20:46:18Z" level=info msg="Starting CRI-O, version: 1.20.1-2.rhaos4.7.git0c3216a.el8, git: ()"
Mar 11 20:46:18 hostname crio[13892]: time="2021-03-11 20:46:18.674669715Z" level=info msg="Version file /var/run/crio/version not found: open /var/run/crio/version: no such file or directory: triggering wipe of containers"
Mar 11 20:46:18 hostname crio[13892]: time="2021-03-11 20:46:18.674688465Z" level=info msg="Version file /var/lib/crio/version not found: open /var/lib/crio/version: no such file or directory: triggering wipe of images"
Mar 11 20:46:18 hostname crio[13892]: time="2021-03-11 20:46:18.675612190Z" level=info msg="Not using native diff for overlay, this may cause degraded performance for building images: kernel has CONFIG_OVERLAY_FS_REDIRECT_DIR enabled"
Mar 11 20:46:18 hostname systemd[1]: crio-wipe.service: Succeeded.
Mar 11 20:46:18 hostname systemd[1]: Started CRI-O Auto Update Script.
Mar 11 20:46:18 hostname systemd[1]: Starting Open Container Initiative Daemon...
Mar 11 20:46:18 hostname crio[13914]: time="2021-03-11T20:46:18Z" level=info msg="Starting CRI-O, version: 1.20.1-2.rhaos4.7.git0c3216a.el8, git: ()"
Mar 11 20:46:18 hostname crio[13914]: time="2021-03-11 20:46:18.729133771Z" level=info msg="Node configuration value for hugetlb cgroup is true"
Mar 11 20:46:18 hostname crio[13914]: time="2021-03-11 20:46:18.729145750Z" level=info msg="Node configuration value for pid cgroup is true"
Mar 11 20:46:18 hostname crio[13914]: time="2021-03-11 20:46:18.729154784Z" level=info msg="Node configuration value for memoryswap cgroup is true"
Mar 11 20:46:18 hostname crio[13914]: time="2021-03-11 20:46:18.735431401Z" level=info msg="Node configuration value for systemd CollectMode is true"
Mar 11 20:46:18 hostname crio[13914]: time="2021-03-11 20:46:18.735794254Z" level=info msg="Not using native diff for overlay, this may cause degraded performance for building images: kernel has CONFIG_OVERLAY_FS_REDIRECT_DIR enabled"
Mar 11 20:46:18 hostname crio[13914]: time="2021-03-11 20:46:18.735876527Z" level=info msg="Using default capabilities: CAP_CHOWN, CAP_DAC_OVERRIDE, CAP_FSETID, CAP_FOWNER, CAP_SETGID, CAP_SETUID, CAP_SETPCAP, CAP_NET_BIND_SERVICE, CAP_KILL"
Mar 11 20:46:18 hostname crio[13914]: time="2021-03-11 20:46:18.791995720Z" level=fatal msg="Validating runtime config: conmon validation: invalid conmon path: stat /usr/libexec/crio/conmon: no such file or directory"
Mar 11 20:46:18 hostname systemd[1]: crio.service: Main process exited, code=exited, status=1/FAILURE
Mar 11 20:46:18 hostname systemd[1]: crio.service: Failed with result 'exit-code'.
Mar 11 20:46:18 hostname systemd[1]: Failed to start Open Container Initiative Daemon.
```

**Comment by haircommander** [comment]
sorry, where exactly did you find it? the current approach to install to centos stream is from OBS (described in our [install docs](github.com/cri-o/cri-o/tree/master/install.md)

**Comment by R37ribution** [comment]
@haircommander, I pulled down `cri-o-1.20.1-2.rhaos4.7.git0c3216a.el8.x86_64.rpm` from the "Red Hat OpenShift Container Platform 4.7 for RHEL 8 x86_64" repository.

I will try the OBS repo Monday and report back.

I appreciate the quick response!

**Comment by haircommander** [comment]
did OBS work for you @R37ribution ?

**Comment by R37ribution** [comment]
@haircommander I tested both the CentOS 8-stream - which the latest version is v1.20.0 and CentOS 8 which has v1.20.1.

I'm curioius why two different RPMs for 8 and 8-Stream. I guess it kind of makes sense, but why is there no v1.20.1 for CentOS 8-Stream?

Either way, I didn't see the issue in either OBS versions. So I'll close the issue.

- CentOS 8-Stream - `v1.20.0`: https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable:/cri-o:/1.20/CentOS_8_Stream/x86_64/cri-o-1.20.0-6.3.el8.x86_64.rpm
  ```
  Mar 19 21:37:26 ctv-mobile-strtus-m-91-05.vi.comcast.net systemd[1]: Starting Container Runtime Interface for OCI (CRI-O)...
  Mar 19 21:37:26 ctv-mobile-strtus-m-91-05.vi.comcast.net crio[23349]: time="2021-03-19T21:37:26Z" level=info msg="Starting CRI-O, version: 1.20.0, git: d388528dbed26b93c5bc1c89623607a1e597aa57(clean)"
  Mar 19 21:37:26 ctv-mobile-strtus-m-91-05.vi.comcast.net crio[23349]: time="2021-03-19 21:37:26.367923036Z" level=info msg="Node configuration value for hugetlb cgroup is true"
  Mar 19 21:37:26 ctv-mobile-strtus-m-91-05.vi.comcast.net crio[23349]: time="2021-03-19 21:37:26.367935724Z" level=info msg="Node configuration value for pid cgroup is true"
  Mar 19 21:37:26 ctv-mobile-strtus-m-91-05.vi.comcast.net crio[23349]: time="2021-03-19 21:37:26.367947864Z" level=info msg="Node configuration value for memoryswap cgroup is true"
  Mar 19 21:37:26 ctv-mobile-strtus-m-91-05.vi.comcast.net crio[23349]: time="2021-03-19 21:37:26.372065045Z" level=info msg="Node configuration value for systemd CollectMode is true"
  Mar 19 21:37:26 ctv-mobile-strtus-m-91-05.vi.comcast.net crio[23349]: time="2021-03-19 21:37:26.497107487Z" level=info msg="Not using native diff for overlay, this may cause degraded performance for building images: kernel has CONFIG_OV>
  Mar 19 21:37:26 ctv-mobile-strtus-m-91-05.vi.comcast.net crio[23349]: time="2021-03-19 21:37:26.497270812Z" level=info msg="Using default capabilities: CAP_CHOWN, CAP_DAC_OVERRIDE, CAP_FSETID, CAP_FOWNER, CAP_SETGID, CAP_SETUID, CAP_SET>
  Mar 19 21:37:26 ctv-mobile-strtus-m-91-05.vi.comcast.net crio[23349]: time="2021-03-19 21:37:26.515981036Z" level=info msg="Conmon does support the --sync option"
  Mar 19 21:37:26 ctv-mobile-strtus-m-91-05.vi.comcast.net crio[23349]: time="2021-03-19 21:37:26.516028669Z" level=info msg="No seccomp profile specified, using the internal default"
  Mar 19 21:37:26 ctv-mobile-strtus-m-91-05.vi.comcast.net crio[23349]: time="2021-03-19 21:37:26.516035234Z" level=info msg="AppArmor is disabled by the system or at CRI-O build-time"
  Mar 19 21:37:26 ctv-mobile-strtus-m-91-05.vi.comcast.net crio[23349]: time="2021-03-19 21:37:26.518546513Z" level=info msg="Found CNI network crio (type=bridge) at /etc/cni/net.d/100-crio-bridge.conf"
  Mar 19 21:37:26 ctv-mobile-strtus-m-91-05.vi.comcast.net crio[23349]: time="2021-03-19 21:37:26.520851563Z" level=info msg="Found CNI network 200-loopback.conf (type=loopback) at /etc/cni/net.d/200-loopback.conf"
  Mar 19 21:37:26 ctv-mobile-strtus-m-91-05.vi.comcast.net crio[23349]: time="2021-03-19 21:37:26.520868551Z" level=info msg="Update default CNI network name to crio"
  Mar 19 21:37:26 ctv-mobile-strtus-m-91-05.vi.comcast.net crio[23349]: W0319 21:37:26.523965   23349 hostport_manager.go:71] The binary conntrack is not installed, this can cause failures in network connection cleanup.
  Mar 19 21:37:26 ctv-mobile-strtus-m-91-05.vi.comcast.net crio[23349]: W0319 21:37:26.526657   23349 hostport_manager.go:71] The binary conntrack is not installed, this can cause failures in network connection cleanup.
  Mar 19 21:37:26 ctv-mobile-strtus-m-91-05.vi.comcast.net systemd[1]: Started Container Runtime Interface for OCI (CRI-O).
  Mar 19 21:37:43 ctv-mobile-strtus-m-91-05.vi.comcast.net systemd[1]: Stopping Container Runtime Interface for OCI (CRI-O)...
  Mar 19 21:37:43 ctv-mobile-strtus-m-91-05.vi.comcast.net systemd[1]: crio.service: Succeeded.
  Mar 19 21:37:43 ctv-mobile-strtus-m-91-05.vi.comcast.net systemd[1]: Stopped Container Runtime Interface for OCI (CRI-O).
  ```

- CentOS 8 - `v1.20.0`: https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable:/cri-o:/1.20/CentOS_8/x86_64/cri-o-1.20.1-4.1.el8.x86_64.rpm
  ```
  Mar 19 21:39:41 ctv-mobile-strtus-m-91-05.vi.comcast.net systemd[1]: Starting Container Runtime Interface for OCI (CRI-O)...
  Mar 19 21:39:41 ctv-mobile-strtus-m-91-05.vi.comcast.net crio[24513]: time="2021-03-19T21:39:41Z" level=info msg="Starting CRI-O, version: 1.20.1, git: 0e6266bc8b26e7f8c1b85df3af7af1dcb50ce813(clean)"
  Mar 19 21:39:41 ctv-mobile-strtus-m-91-05.vi.comcast.net crio[24513]: time="2021-03-19 21:39:41.484459110Z" level=info msg="Node configuration value for hugetlb cgroup is true"
  Mar 19 21:39:41 ctv-mobile-strtus-m-91-05.vi.comcast.net crio[24513]: time="2021-03-19 21:39:41.484470500Z" level=info msg="Node configuration value for pid cgroup is true"
  Mar 19 21:39:41 ctv-mobile-strtus-m-91-05.vi.comcast.net crio[24513]: time="2021-03-19 21:39:41.484480152Z" level=info msg="Node configuration value for memoryswap cgroup is true"
  Mar 19 21:39:41 ctv-mobile-strtus-m-91-05.vi.comcast.net crio[24513]: time="2021-03-19 21:39:41.490678508Z" level=info msg="Node configuration value for systemd CollectMode is true"
  Mar 19 21:39:41 ctv-mobile-strtus-m-91-05.vi.comcast.net crio[24513]: time="2021-03-19 21:39:41.491050285Z" level=info msg="Not using native diff for overlay, this may cause degraded performance for building images: kernel has CONFIG_OV>
  Mar 19 21:39:41 ctv-mobile-strtus-m-91-05.vi.comcast.net crio[24513]: time="2021-03-19 21:39:41.491135812Z" level=info msg="Using default capabilities: CAP_CHOWN, CAP_DAC_OVERRIDE, CAP_FSETID, CAP_FOWNER, CAP_SETGID, CAP_SETUID, CAP_SET>
  Mar 19 21:39:41 ctv-mobile-strtus-m-91-05.vi.comcast.net crio[24513]: time="2021-03-19 21:39:41.557130351Z" level=info msg="Conmon does support the --sync option"
  Mar 19 21:39:41 ctv-mobile-strtus-m-91-05.vi.comcast.net crio[24513]: time="2021-03-19 21:39:41.557206994Z" level=info msg="No seccomp profile specified, using the internal default"
  Mar 19 21:39:41 ctv-mobile-strtus-m-91-05.vi.comcast.net crio[24513]: time="2021-03-19 21:39:41.557213832Z" level=info msg="AppArmor is disabled by the system or at CRI-O build-time"
  Mar 19 21:39:41 ctv-mobile-strtus-m-91-05.vi.comcast.net crio[24513]: time="2021-03-19 21:39:41.559645003Z" level=info msg="Found CNI network crio (type=bridge) at /etc/cni/net.d/100-crio-bridge.conf"
  Mar 19 21:39:41 ctv-mobile-strtus-m-91-05.vi.comcast.net crio[24513]: time="2021-03-19 21:39:41.562025617Z" level=info msg="Found CNI network 200-loopback.conf (type=loopback) at /etc/cni/net.d/200-loopback.conf"
  Mar 19 21:39:41 ctv-mobile-strtus-m-91-05.vi.comcast.net crio[24513]: time="2021-03-19 21:39:41.562041004Z" level=info msg="Update default CNI network name to crio"
  Mar 19 21:39:41 ctv-mobile-strtus-m-91-05.vi.comcast.net crio[24513]: W0319 21:39:41.564954   24513 hostport_manager.go:71] The binary conntrack is not installed, this can cause failures in network connection cleanup.
  Mar 19 21:39:41 ctv-mobile-strtus-m-91-05.vi.comcast.net crio[24513]: W0319 21:39:41.567454   24513 hostport_manager.go:71] The binary conntrack is not installed, this can cause failures in network connection cleanup.
  Mar 19 21:39:41 ctv-mobile-strtus-m-91-05.vi.comcast.net systemd[1]: Started Container Runtime Interface for OCI (CRI-O).
  Mar 19 21:43:00 ctv-mobile-strtus-m-91-05.vi.comcast.net systemd[1]: Stopping Container Runtime Interface for OCI (CRI-O)...
  Mar 19 21:43:00 ctv-mobile-strtus-m-91-05.vi.comcast.net systemd[1]: crio.service: Succeeded.
  Mar 19 21:43:00 ctv-mobile-strtus-m-91-05.vi.comcast.net systemd[1]: Stopped Container Runtime Interface for OCI (CRI-O).
  ```

**Comment by haircommander** [comment]
I actually don't know why there are separate versions (cc @lsm5 ), but I'm glad you got it working!

**Comment by lsm5** [comment]
> I actually don't know why there are separate versions (cc @lsm5 ), but I'm glad you got it working!

@haircommander https://build.opensuse.org/package/show/devel:kubic:libcontainers:stable:cri-o:1.20/cri-o ... looks like the stream build failed, so until that happens, it will continue to ship the last successful build. Does that help?

**Comment by haircommander** [comment]
oh lol I misread the question, I thought it was why two different packages (and thus didn't look at the failures). I'll TAL at that haha

**Comment by R37ribution** [comment]
Hey guys, thanks for the replies. @lsm5 @haircommander is there anything that can be done to get the new build v1.20.1 to complete for 8-Stream? I noticed that v1.20.2 is suppose to address a CAP_PERFMON issue that we're seeing #4658

**Comment by haircommander** [comment]
> is there anything that can be done to get the new build v1.20.1 to complete for 8-Stream

working on it :) 
> would this build issue need to be resolved to get v1.20.2 to build correctly for 8-Stream

I'm not sure, 1.20.2 has hit some snags and is delayed a bit. I'll make sure it makes it in once released :)

**Comment by haircommander** [comment]
I may have figured it out!
